src/components/tables/RowActions/Transform.tsx

```diff
@@ -0,0 +1,61 @@
+import { Button } from '@mui/material';
+import { authenticatedRoutes } from 'app/routes';
+import { useZustandStore } from 'context/Zustand/provider';
+import { GlobalSearchParams } from 'hooks/searchParams/useGlobalSearchParams';
+import { FormattedMessage } from 'react-intl';
+import { useNavigate } from 'react-router';
+import { SelectTableStoreNames } from 'stores/names';
+import {
+    SelectableTableStore,
+    selectableTableStoreSelectors,
+} from 'stores/Tables/Store';
+import { getPathWithParams } from 'utils/misc-utils';
+
+interface Props {
+    selectableTableStoreName: SelectTableStoreNames.COLLECTION;
+}
+
+function Transform({ selectableTableStoreName }: Props) {
+    const navigate = useNavigate();
+
+    const selectedRows = useZustandStore<
+        SelectableTableStore,
+        SelectableTableStore['selected']
+    >(selectableTableStoreName, selectableTableStoreSelectors.selected.get);
+
+    const hasSelections = selectedRows.size > 0;
```

Comment:
I will leave it up to you

We have duplicated this so many times - should we make a hook? I am like split on this cause the hook will be **tiny** and possibly not worth the effort.

Reply:
Maybe the better approach is to make a hook for the `RowActions` to all share in general so we have a "headless component" kind of thing.

Reply:
Funnily enough, it started as a hook which was fairly small. I meant to leave a comment in this PR calling out the duplication and getting a temperature check for how broadly you were comfortable sharing code at this time.